### PR TITLE
PLENTY-133 Support landing pages

### PR DESCRIPTION
**Pull request by darius-adeoweb**

### CODE REVIEW

File: tests/Api/Response/ResponseParserTest.php [hunk 1]
```diff
@@ -39,12 +39,19 @@ public function setUp()
         $this->loggerFactory->expects($this->any())->method('getLogger')->willReturn($this->logger);
     }
 
+    /**
+     * @runInSeparateProcess
```
Comment (ihanli) [inline tests/Api/Response/ResponseParserTest.php]: Why is this necessary?

File: src/Api/Response/ResponseParser.php [hunk 2]
```diff
@@ -41,9 +41,9 @@ public function parse($responseData)
 
         try {
             $data = $this->loadXml($responseData);
+            $this->handleLandingPage($data);
```
Comment (ihanli) [inline src/Api/Response/ResponseParser.php]: For better testability I would suggest:
* Call `parseLandingPage`.
* Pass the return value (if valid) to `handleLandingPage`.
* Instead of mocking `parseLandingPage` in the test, check if `handleLandingPage` has been called with the correct parameter.

File: src/Api/Response/ResponseParser.php [hunk 3]
```diff
@@ -123,7 +134,7 @@ protected function parseLandingPage(\SimpleXMLElement $data)
     {
         $landingPage = [];
 
-        if (!empty($data->landingPage) ) {
+        if (isset($data->landingPage) && !empty($data->landingPage->attributes()) ) {
```
Comment (ihanli) [inline src/Api/Response/ResponseParser.php]: Since there is no apparent reason to return an array (anymore): Return the link as string or `null` if no link was provided.

File: tests/Api/Response/ResponseParserTest.php [hunk 4]
```diff
@@ -39,12 +39,19 @@ public function setUp()
...
+     */
     public function testParse()
     {
         $responseMock = $this->getMockBuilder(Response::class)->disableOriginalConstructor()->setMethods(null)->getMock();
         /** @var ResponseParser|MockObject $responseParserMock */
-        $responseParserMock = $this->getResponseParserMock(['createResponseObject']);
+        $responseParserMock = $this->getResponseParserMock(['createResponseObject', 'parseLandingPage']);
         $responseParserMock->expects($this->any())->method('createResponseObject')->willReturn($responseMock);
+        $responseParserMock->expects($this->once())
```
Comment (ihanli) [inline tests/Api/Response/ResponseParserTest.php]: Check my previous comment regarding testability.

File: tests/Api/Response/ResponseParserTest.php (hunk 1 above)
Comment (darius-adeoweb) [reply tests/Api/Response/ResponseParserTest.php]: It fails the test without it. The problem is with location header being added. 
Found a fix with process isolation here: https://github.com/sebastianbergmann/phpunit/issues/720

File: src/Api/Response/ResponseParser.php (hunk 3 above)
Comment (ihanli) [reply src/Api/Response/ResponseParser.php]: This method still returns an array when it actually should return the string representation of `$data->landingPage->attributes()->link` or `null`. Don't forget to update the tests and PHPDocs.
